src/main/webapp/gast/person.jsp

```diff
@@ -1,11 +1,34 @@
+<%@page import="de.uni_tuebingen.ub.nppm.model.Einzelbeleg"%>
 <%@ page import="de.uni_tuebingen.ub.nppm.util.Language" isThreadSafe="false" %>
+<%@ page import="de.uni_tuebingen.ub.nppm.exception.*" isThreadSafe="false" %>
 <%@ include file="../configuration.jsp"%>
 <%@ include file="../functions.jsp" %>
 
 <jsp:include page="../dofilter.jsp" />
 
-<%
-   int id = Integer.parseInt(request.getParameter("ID"));
+<%    int id = Integer.parseInt(request.getParameter("ID"));
+
+    if(PersonDB.getById(id) == null){
+         throw new IdNotFoundException("Person ID ist nicht vorhanden");
+    }
+
+    List<Einzelbeleg> listEinzelbeleg = EinzelbelegDB.getListEinzelbelegeByPersonId(id);
```

Comment:
Statt hier eine neue Funktion zu schreiben wäre es besser die Model class zu verwenden. Das wird ja oben ohnehin schon abgefragt aber keiner Variable zugewiesen.

```
Person person = Person.getById(id);
if (person == null) {
    throw new IdNotFoundException("Person ID ist nicht vorhanden");
}
List<Einzelbeleg> listEinzelbeleg = person.getEinzelbeleg();
```

Person.getEinzelbeleg() gibt es bereits (evtl sollte man sie in getEinzelbelege() umbenennen da sie ja mehrere zurück gibt.

Reply:
done